### PR TITLE
Add pagination and tests to Clients and Connections endpoints

### PR DESCRIPTION
**Pull request by joshcanhelp**
- Add new `:page` and `:per_page` method parameters to `Auth0::Api::V2::Clients` and `Auth0::Api::V2::Connections` modules.
- Add ability to pass an `Array` for the `fields:` parameter and convert to comma-sep string
- Add tests for all changes above.

### CODE REVIEW

File: spec/lib/auth0/api/v2/clients_spec.rb [hunk 1]
```diff
@@ -5,18 +5,62 @@
...
+        fields: 'name,app_type',
+        page: nil,
+        per_page: nil
+      )
+      expect {
+        @instance.clients(fields: ['name','app_type'], include_fields: true)
+      }.not_to raise_error
+    end
+
+    it 'is expected to add pagination' do
```
Comment (machuga) [inline spec/lib/auth0/api/v2/clients_spec.rb]: This is asserting something about adding pagination, but only tests that an error wasn't raised. Is there an indicator that the params weren't silently ignored?

Comment (joshcanhelp) [reply spec/lib/auth0/api/v2/clients_spec.rb]: So ... that first `expect` line sets up the second one. If you change the `page` value in `with()` to a different value, it fails.

**Comment by machuga** [comment]
Right, but I'm suggesting that code wise I cannot tell that what is being
tested is ensuring pagination works.

**Comment by joshcanhelp** [comment]
@machuga - As in, you can't tell if the results themselves are paginated? I can add an integration test for that. 

But, for a unit test, I *think* this works ok. It was a bit of trial and error to get it to work but changing the values of, say, the `page` parameter on one or the other but not both would make the test fail. So that first `expect` is saying "make the instance get an endpoint with the following parameters" and the second calls that instance, causing an error if it does not meet the expectation. [Source](https://relishapp.com/rspec/rspec-mocks/docs/basics/expecting-messages).

**Comment by machuga** [comment]
Ya I think the test name just implies that something else is being tested. `it ......`

**Pull request merged**